isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -3033,6 +3027,49 @@ namespace isobus
 							}
 						}
 						break;
+
+						case static_cast<std::uint8_t>(Function::HideShowObjectCommand):
+						case static_cast<std::uint8_t>(Function::EnableDisableObjectCommand):
+						case static_cast<std::uint8_t>(Function::SelectInputObjectCommand):
+						case static_cast<std::uint8_t>(Function::ESCCommand):
+						case static_cast<std::uint8_t>(Function::ControlAudioSignalCommand):
+						case static_cast<std::uint8_t>(Function::SetAudioVolumeCommand):
+						case static_cast<std::uint8_t>(Function::ChangeChildLocationCommand):
+						case static_cast<std::uint8_t>(Function::ChangeChildPositionCommand):
+						case static_cast<std::uint8_t>(Function::ChangeSizeCommand):
+						case static_cast<std::uint8_t>(Function::ChangeBackgroundColourCommand):
+						case static_cast<std::uint8_t>(Function::ChangeNumericValueCommand):
+						case static_cast<std::uint8_t>(Function::ChangeStringValueCommand):
+						case static_cast<std::uint8_t>(Function::ChangeEndPointCommand):
+						case static_cast<std::uint8_t>(Function::ChangeFontAttributesCommand):
+						case static_cast<std::uint8_t>(Function::ChangeLineAttributesCommand):
+						case static_cast<std::uint8_t>(Function::ChangeFillAttributesCommand):
+						case static_cast<std::uint8_t>(Function::ChangeActiveMaskCommand):
+						case static_cast<std::uint8_t>(Function::ChangeSoftKeyMaskCommand):
+						case static_cast<std::uint8_t>(Function::ChangeAttributeCommand):
+						case static_cast<std::uint8_t>(Function::ChangePriorityCommand):
+						case static_cast<std::uint8_t>(Function::ChangeListItemCommand):
+						case static_cast<std::uint8_t>(Function::DeleteObjectPoolCommand):
+						case static_cast<std::uint8_t>(Function::LockUnlockMaskCommand):
+						case static_cast<std::uint8_t>(Function::ExecuteMacroCommand):
+						case static_cast<std::uint8_t>(Function::ChangeObjectLabelCommand):
+						case static_cast<std::uint8_t>(Function::ChangePolygonPointCommand):
+						case static_cast<std::uint8_t>(Function::ChangePolygonScaleCommand):
+						case static_cast<std::uint8_t>(Function::GraphicsContextCommand):
+						case static_cast<std::uint8_t>(Function::GetAttributeValueMessage):
+						case static_cast<std::uint8_t>(Function::SelectColourMapCommand):
+						case static_cast<std::uint8_t>(Function::ExecuteExtendedMacroCommand):
+						case static_cast<std::uint8_t>(Function::SelectActiveWorkingSet):
+						{
+							// By checking if it's a response with our control functions, we verify that it's a response to a request we sent.
+							// This because we only support Working Set Masters at the moment.
+							if ((parentVT->myControlFunction == message.get_destination_control_function()) &&
+							    (parentVT->partnerControlFunction == message.get_source_control_function()))
+							{
+								parentVT->commandAwaitingResponse = false;
```

Comment:
If you take my other suggestion we may also want to reset the timestamp to 0 here

Reply:
With my edited solution we don't need to set timestamp to 0, I think it's valid to always leave it at the timestamp the last command was sent.